src/qibo/models/encodings.py

```diff
@@ -73,6 +74,86 @@ def unary_encoder(data):
+    """Creates a circuit that performs the unary encoding of a random Gaussian state.
+
+    Given :math:`d` qubits, encodes the quantum state
+    :math:`\\ket{\\psi} \\in \\mathcal{H}` such that
+
+
+    .. math::
+        \\ket{\\psi} = \\frac{1}{\\|\\mathbf{x}\\|_{\\textup{HS}}} \\,
+            \\sum_{k=1}^{d} \\, x_{k} \\, \\ket{k}
+
+    where :math:`x_{k}` are independent Gaussian random variables,
+    :math:`\\mathcal{H} \\cong \\mathbb{C}^{d}` is a :math:`d`-qubit Hilbert space,
+    and :math:`\\|\\cdot\\|_{\\textup{HS}}` being the Hilbert-Schmidt norm.
+    Here, :math:`\\ket{k}` is a unary representation of the number :math:`1` through
+    :math:`d`.
+
+    At depth :math:`h`, the angles :math:`\\theta_{k} \\in [0, 2\\pi]` of the the
+    gates :math:`RBS(\\theta_{k})` are sampled from the following probability density function:
+
+    .. math::
+        p_{h}(\\theta) = \\frac{1}{2} \\, \\frac{\\Gamma(2^{h-1})}{\\Gamma^{2}(2^{h-2})}
+            \\abs{\\sin(\\theta) \\, \\cos(\\theta)}^{2^{h-1} - 1} \\, ,
+
+    where :math:`\\Gamma(\\cdot)` is the
+    `Gamma function <https://en.wikipedia.org/wiki/Gamma_function>`_.
+
+    Args:
+        nqubits (int): number of qubits.
+        seed (int or :class:`numpy.random.Generator`, optional): Either a generator of
+            random numbers or a fixed seed to initialize a generator. If ``None``,
+            initializes a generator with a random seed. Defaults to ``None``.
+
+    Returns:
+        :class:`qibo.models.circuit.Circuit`: circuit that loads a random Gaussian array in unary representation.
+
+    References:
+        1. A. Bouland, A. Dandapani, and A. Prakash, *A quantum spectral method for simulating
+        stochastic processes, with applications to Monte Carlo*.
+        `arXiv:2303.06719v1 [quant-ph] <https://arxiv.org/abs/2303.06719>`_
+    """
+    if not isinstance(nqubits, int):
+        raise_error(
+            TypeError, f"nqubits must be type int, but it is type {type(nqubits)}."
+        )
+    elif nqubits <= 0.0:
+        raise_error(
+            ValueError, f"nqubits must be a positive integer, but it is {nqubits}."
+        )
+    elif not math.log2(nqubits).is_integer():
+        raise_error(ValueError, f"nqubits must be a power of 2, but it is {nqubits}.")
+
+    if (
+        seed is not None
+        and not isinstance(seed, int)
+        and not isinstance(seed, np.random.Generator)
+    ):
+        raise_error(
+            TypeError, "seed must be either type int or numpy.random.Generator."
+        )
+
+    local_state = (
+        np.random.default_rng(seed) if seed is None or isinstance(seed, int) else seed
+    )
+
+    sampler = _probability_distribution_gaussian_loader(
+        a=0, b=2 * math.pi, seed=local_state
+    )
+
+    circuit, pairs_rbs = _generate_rbs_pairs(nqubits)
+
+    phases = []
+    for depth, row in enumerate(pairs_rbs, 1):
+        phases += list(sampler.rvs(depth=depth, size=len(row)))
```

Comment:
Usually
```suggestion
        phases.extend(sampler.rvs(depth=depth, size=len(row)))
```
is better because it extends the existing list (while `+` recreates a new list every time).